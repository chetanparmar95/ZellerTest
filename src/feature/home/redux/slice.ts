import {createSlice} from '@reduxjs/toolkit';

export interface ZellerCustomer {
  id: string;
  name?: string;
  email?: string;
  role?: string;
}

interface ZellerCustomerConnection {
  items: ZellerCustomer[];
  nextToken?: string;
}

export type UserState = {
  customers?: ZellerCustomerConnection;
};

export const initialState: UserState = {};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    getUsers(_state, _action) {},
    setUsers(state, action) {
      state.customers = action.payload;
    },
  },
});

export const {getUsers, setUsers} = userSlice.actions;

export default userSlice.reducer;
