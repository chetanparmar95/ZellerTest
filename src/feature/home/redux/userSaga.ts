import {call, put, takeLatest} from 'redux-saga/effects';
import {getUserList} from '../../../services/userService';
import {getUsers, setUsers} from './slice';

export function* fetchUsersSaga(action: any): Generator<any> {
  try {
    const userData: any = yield call(getUserList, action.payload);
    yield put(setUsers(userData));
  } catch (error) {
    console.log(error);
  }
}

function* userSaga() {
  yield takeLatest(getUsers.type, fetchUsersSaga);
}

export default userSaga;
