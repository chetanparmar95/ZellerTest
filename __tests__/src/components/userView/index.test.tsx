import 'react-native';
import React from 'react';
import UserView from '../../../../src/components/userView';
import {ZellerCustomer} from '../../../../src/feature/home/redux/slice';
import renderer from 'react-test-renderer';

describe('renders correctly', () => {
  test('Render init', () => {
    const user: ZellerCustomer = {
      id: '1',
      name: 'David Miller',
      email: 'david@gmail.com',
      role: 'ADMIN',
    };
    expect(renderer.create(<UserView user={user} />)).toMatchSnapshot();
  });
});
