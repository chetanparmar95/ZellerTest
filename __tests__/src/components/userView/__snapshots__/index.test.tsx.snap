// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Render init 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 4,
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(234,242,250)",
        "borderRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(48,111,199)",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      D
    </Text>
  </View>
  <View
    style={
      {
        "marginHorizontal": 16,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
        }
      }
    >
      David Miller
    </Text>
    <Text
      style={
        {
          "color": "gray",
          "fontSize": 12,
          "marginTop": 4,
        }
      }
    >
      ADMIN
    </Text>
  </View>
</View>
`;
